### PR TITLE
[DESK-397] device name add validation

### PR DESCRIPTION
**Pull request by mauricio0000**
### Description

Validation of the device limit max and reuse the same validation

### Checklist

- [x] Pull Request title is in the format `[PROJ-12345] Some useful description here...`
- [x] I have prefixed this PR with `[WIP]` if it is a Work In Progress (not ready to merge)
- [x] The git branch is in the format `PROJ-12345-some-useful-description-here`
- [x] I have added one or more reviewers
- [x] I have added/updated tests for any code changes
- [x] I have updated documentation (including README, inline comments and docs.remote.it docs)
- [x] This PR only contains one isolated change (bug fix, feature, etc)
- [x] Translation strings added/updated for all user-visible text
- [x] I have manually reviewed this PR to make sure only the files I intended to change were changed and nothing else
- [x] I have notified reviewers on Slack by @ mentioning them in the `#desktop`channel
- [ ] I have moved the Jira ticket into `Resolved` state and made a note in the ticket with a link to this PR

### Test plan

1. Sign in
2. Go to Add a device
3. Try to add characters nonalphanumeric or name more than 100 characters
4. You should see the validation

### Ticket

https://remoteit.atlassian.net/browse/DESK-397

### CODE REVIEW

File: backend/src/sharedCopy/nameHelper.ts [hunk 1]
```diff
@@ -66,3 +66,20 @@ export function osName(os?: Ios) {
...
+      value
+    }
+  }
+  if (value.length > 100) {
+    return  { 
+      error: 'Device names has limit off 100 characters.',
+      value: value.substring(0, 100)
+    }
+  }
+  return {value}
```
Comment (JamieRuderman) [inline backend/src/sharedCopy/nameHelper.ts:84]: Looks good, but please use the https://prettier.io/ code formatter on save.

File: frontend/src/components/InlineSetting/InlineSetting.tsx [hunk 2]
```diff
@@ -73,7 +75,14 @@ export const InlineSetting: React.FC<Props> = ({
             label={label}
             value={editValue}
             variant="filled"
-            onChange={event => setEditValue(filter ? event.target.value.replace(filter, '') : event.target.value)}
+            helperText={nameError || ''}
```
Comment (JamieRuderman) [inline frontend/src/components/InlineSetting/InlineSetting.tsx:78]: This is the wrong component to add the validation to.

**Comment by mauricio0000** [comment]
@JamieRuderman I have moved the conditions of MAX_NAME_LENGTH to https://github.com/remoteit/desktop/pull/170/files#diff-c5aaf389cd2c575ac4c99b5fcd7e32d2R82 because there is reusing the condition.

- setName(value) is working in the result of the validation https://github.com/remoteit/desktop/pull/170/files#diff-9a3d4257b53fe8425aa693a6fe8d2f2aR78

image

**Comment by mauricio0000** [comment]
Screen Shot 2020-06-30 at 11 11 05

File: frontend/src/shared/nameHelper.ts [hunk 3]
```diff
@@ -70,3 +70,20 @@ export function osName(os?: Ios) {
...
+  const value = name.replace(REGEX_NAME_SAFE, '')
+  if (value !== name) {
+    return  { 
+      error: 'Device names can only contain alpha numeric characters.',
+      value
+    }
+  }
+  if (validateLength && value.length > MAX_NAME_LENGTH) {
+    return  { 
+      error: `Device names cannot exceed ${MAX_NAME_LENGTH} characters.`,
```
Comment (JamieRuderman) [inline frontend/src/shared/nameHelper.ts:84]: Please
```suggestion
      error: `Cannot exceed ${MAX_NAME_LENGTH} characters.`,
```

File: frontend/src/shared/nameHelper.ts [hunk 4]
```diff
@@ -70,3 +70,20 @@ export function osName(os?: Ios) {
 export function safeFilename(name: string) {
   return name.replace(REGEX_NOT_FILE_SAFE, '-')
 }
+
+export function serviceNameValidation(name: string, validateLength?: boolean) {
+  const value = name.replace(REGEX_NAME_SAFE, '')
+  if (value !== name) {
+    return  { 
+      error: 'Device names can only contain alpha numeric characters.',
```
Comment (JamieRuderman) [inline frontend/src/shared/nameHelper.ts:78]: Sorry missed this reference too:
```suggestion
      error: 'Can only contain alpha numeric characters.',
```

**Pull request merged**